app/views/auth/sessions/two_factor.html.haml

```diff
@@ -1,14 +1,9 @@
 - content_for :page_title do
   = t('auth.login')
 
-= simple_form_for(resource, as: resource_name, url: session_path(resource_name), method: :post) do |f|
-  %p.hint.otp-hint= t('simple_form.hints.sessions.otp')
+=javascript_pack_tag 'two_factor_authentication', integrity: true, crossorigin: 'anonymous'
 
-  .fields-group
-    = f.input :otp_attempt, type: :number, wrapper: :with_label, label: t('simple_form.labels.defaults.otp_attempt'), input_html: { 'aria-label' => t('simple_form.labels.defaults.otp_attempt'), :autocomplete => 'off' }, autofocus: true
+- if @webauthn_enabled
+  = render partial: 'auth/sessions/two_factor/webauthn_form', locals: { hidden: @scheme_type != 'webauthn' }
```

Comment:
Is it important to render this if you're gonna hide it?

Reply:
Yes! The thing is that when you are attempting to log in and you have `WebAuthn` enabled, you will land on a page to use your security keys to log in, but if you want to log in using an OTP code from your phone you have to be able to do so, and that's why we added that link below in the last GIF: *Don't you have your security key? Enter a two-factor code from your phone or a recovery code*. That link will render the page for using an OTP code as second factor authentication.

In the current implementation, when users hit `create` from `SessionsController`, if they have second factor enabled, the action will render this `two_factor` view. So, suppose that instead of that view we have two: `otp_2fa_form` and `webauthn_2fa_form`, the problem is that if we wanted to change between them using a link, we would have to hit `SessionsController#create` again with user's username and password and render, which is not possible.

As we couldn't find another way of doing it, we decided of introduce the code of the two forms in the same view and use JavaScript to hide and show one or the other.